### DISCUSSION TITLE
Possible to sort tabs in a tree alphabetically?

### DISCUSSION THREAD

**Discussion by kings2u** [piroor/treestyletab#3149] [Ideas]
I think this would be a killer feature. I picture it only sorting the direct children of the tree you are executing on, not all descendants. Is there an add-on that does this?

**Comment by irvinm** [comment]
TST just shows the tabs in an alternate view.  Any tab sorting addon should do the trick if you want to do it across all of your tabs.  The challenge with any tab sorting addon is how it handles (or mishandles) setting the "openerTabId" which is used to maintain parent-child relationships. 

I am guessing there would need to be a more specific version of one of these types of addons in order to just work on "children of the tree" ... basically be more "tree aware" of existing relationships and what scope to take action on.

I haven't tried any of these, but worth a look:

- https://addons.mozilla.org/en-US/firefox/addon/sort-tabs-advanced/
- https://addons.mozilla.org/en-US/firefox/addon/sort-tabs-by-url/
- https://addons.mozilla.org/en-US/firefox/addon/sorted-tabs/
